tracing/src/instrument.rs

```diff
@@ -136,18 +187,63 @@ pub trait WithCollector: Sized {
         }
     }
 
-    /// Attaches the current [default] collector to this type, returning a
-    /// `WithDispatch` wrapper.
+    /// Attaches the current [default] [collector] to this type, returning a
+    /// [`WithDispatch`] wrapper.
     ///
-    /// When the wrapped type is a future, stream, or sink, the attached
-    /// collector will be set as the [default] while it is being polled.
-    /// When the wrapped type is an executor, the collector will be set as the
-    /// default for any futures spawned on that executor.
+    /// The attached collector will be set as the [default] when the returned
+    /// [`Future`] is polled.
     ///
     /// This can be used to propagate the current dispatcher context when
-    /// spawning a new future.
+    /// spawning a new future that may run on a different thread.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # mod tokio {
+    /// #     pub(super) fn spawn(_: impl std::future::Future) {}
+    /// # }
+    /// # pub struct MyCollector;
+    /// # impl tracing::Collect for MyCollector {
+    /// #   fn new_span(&self, _: &tracing::span::Attributes) -> tracing::span::Id {
+    /// #       tracing::span::Id::from_u64(0)
+    /// #   }
+    /// #   fn record(&self, _: &tracing::span::Id, _: &tracing::span::Record) {}
+    /// #   fn event(&self, _: &tracing::Event<'_>) {}
+    /// #   fn record_follows_from(&self, _: &tracing::span::Id, _: &tracing::span::Id) {}
+    /// #   fn enabled(&self, _: &tracing::Metadata) -> bool { false }
+    /// #   fn enter(&self, _: &tracing::span::Id) {}
+    /// #   fn exit(&self, _: &tracing::span::Id) {}
+    /// #   fn current_span(&self) -> tracing_core::span::Current {
+    /// #       tracing_core::span::Current::unknown()
+    /// #    }
+    /// # }
+    /// # impl MyCollector { fn new() -> Self { Self } }
+    /// # async fn docs() {
+    /// use tracing::instrument::WithCollector;
+    ///
+    /// // Using `set_default` (rather than `set_global_default`) sets the
+    /// // default collector for *this* thread only.
+    /// let _default = tracing::collect::set_default(MyCollector::new());
+    ///
+    /// let future = async {
+    ///     // ...
+    /// };
+    ///
+    /// // If a multi-threaded async runtime is in use, this spawned task may
+    /// // run on a different thread, in a different default collector's context.
+    /// tokio::spawn(future);
     ///
+    /// // However, if we call `with_current_collector` on the future before
+    /// // spawning it, we can ensure that the current thread's default
+    /// // collector is propagated to the spawned task, regardless of where
+    /// // it executes:
```

Comment:
i'd maybe remove the "we" used here—i don't know how much that's used in tracing's docs nowadays.